### ISSUE TITLE
Restart Button Missing

### ISSUE THREAD

**Issue by slumbergod** [BrowserWorks/waterfox#3595]
### What happened?

I just upgraded to Waterfox 6.5.0 in Manjaro Linux.
I was tweaking some css customisations but when I went to restart waterfox from the main menu I noticed there is no restart option, only quit. In my preferences I have it enabled and I have disabled and reenabled it several times and restarted the browser between changes but I cannot get the restart option back.

Is anyone else experiencing this? Had the option moved location from main menu?

### Reproducible?

- [x] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

Other

### What platform are you seeing the problem on?

Linux

**Comment by ghost** [comment]
On Mint 22, I also have no restart button, although the option is set in preferences.

**Comment by aermak** [comment]
On win11 and missing that restart button as well.

**Comment by Bonza13** [comment]
Turn on\off in settings doesn't work.

**Comment by NotesTracker** [comment]
Same problem with Windows 10, 

Reverted to older G6.0.20 version and will await fix. This also fixed separate annoying issue in 6.5.0 where the bottom part of all tabs in the tab bar were being truncated (tab height issue).

**Comment by Splitwirez** [comment]
Same here - thought I was going crazy, started questioning if the option really existed at all or if I'd just misremembered. (using Kubuntu Linux, if it makes any difference)

**Comment by toasterking** [comment]
Me too, with WF 6.5.0 on Win10.

**Comment by jabcreations** [comment]
I have fixed this at my end and am implementing a couple of tutorials in my very limited free time. This will not count as a fix in the browser itself though a user-applied-patch (not an official term) though _it will work_. I'll post an update later once the tutorials are reasonably well established.

![Image](https://github.com/user-attachments/assets/5ddbcd68-dc20-4317-8abd-f5696ccc42b5)

**Comment by jabcreations** [comment]
There are two tutorials to do a user-patched restart browser button. You have to _first enable the userChrome.js_ file and then you can follow the shorter tutorial on adding the restart browser button.

I've created the tutorials available at the Fixed Firefox website here:
https://www.fixedfirefox.com/userchrome-js/

**Comment by robbely** [comment]
> I have fixed this at my end and am implementing a couple of tutorials in my very limited free time. This will not count as a fix in the browser itself though a user-applied-patch (not an official term) though _it will work_. I'll post an update later once the tutorials are reasonably well established.
> 
> ![Image](https://github.com/user-attachments/assets/5ddbcd68-dc20-4317-8abd-f5696ccc42b5)

Can you explain how you get the icons next to the file menu items? I only have blank spaces where these should be, and the new restart item doesn't even have the icon space, just the text.

**Comment by Ketrel** [comment]
It's still missing in 6.5.1 as well.

**Comment by robbely** [comment]
> It's still missing in 6.5.1 as well.

It's listed in "known issues" now though:

[https://www.waterfox.net/docs/releases/6.5.1/](url)

**Comment by Ketrel** [comment]
Oddly enough thanks to https://github.com/BrowserWorks/Waterfox/issues/3610 I didn't get the releases page when the updater ran when I used the LD_LIBRARY_PATH method.  I noticed the restart missing when I went to quit the browser so I could reopen it without the library path override

**Comment by jabcreations** [comment]
> > I have fixed this at my end and am implementing a couple of tutorials in my very limited free time. This will not count as a fix in the browser itself though a user-applied-patch (not an official term) though _it will work_. I'll post an update later once the tutorials are reasonably well established.
> > ![Image](https://github.com/user-attachments/assets/5ddbcd68-dc20-4317-8abd-f5696ccc42b5)
> 
> Can you explain how you get the icons next to the file menu items? I only have blank spaces where these should be, and the new restart item doesn't even have the icon space, just the text.

That is Waterfox, not Firefox though you're here in the Waterfox Git so... be more specific to get more a desirable response.

**Comment by robbely** [comment]
> That is Waterfox, not Firefox though you're here in the Waterfox Git so... be more specific to get more a desirable response.

Yes, I'm using Waterfox. I was referring to the icons in the image to the left of the menu items (eg the envelope to the left of "Email Link" etc), I just have empty spaces where those icons would appear on mine. From other images I've seen of both Waterfox and even recent Firefox, those icons aren't there any more so I was wondering what you used to bring them back.

EDIT: Here is an image showing what I am getting in my File menu:

![Image](https://github.com/user-attachments/assets/fc288034-f4b4-44f5-849a-61173bbe5dee)

**Comment by jabcreations** [comment]
Go to _**about:config**_ and set _**userChrome.icon.global_menubar**_ to **_true_** (and then restart the browser).

**Comment by robbely** [comment]
> Go to _**about:config**_ and set _**userChrome.icon.global_menubar**_ to **_true_** (and then restart the browser).

I checked and it is already set to true. It looks like you need custom css to do it (eg this one [https://gist.github.com/qaz69wsx/83a90423163b65a8344b2e60fc356a58](url) which is incomplete. Perhaps you added something to userchrome.css in the past to do this?

**Comment by jabcreations** [comment]
Are you using Waterfox 6.0 (115 ESR) or 6.5 (128 ESR)? I'm not sure about Waterfox versions after 56 and before 115.

You may have to create a new profile and then port over your tab sessions and files. You can go through my Fixed Firefox profiles page to figure that out here:

https://www.fixedfirefox.com/profiles/

Basically start a new profile and import a few files at a time. If importing something breaks the new profile then you know where the issue is in the old profile.

**Comment by robbely** [comment]
> Are you using Waterfox 6.0 (115 ESR) or 6.5 (128 ESR)? I'm not sure about Waterfox versions after 56 and before 115.
> 
> You may have to create a new profile and then port over your tab sessions and files. You can go through my Fixed Firefox profiles page to figure that out here:
> 
> https://www.fixedfirefox.com/profiles/
> 
> Basically start a new profile and import a few files at a time. If importing something breaks the new profile then you know where the issue is in the old profile.

I am using 6.5 (though I remember I didn't have the icons on 6.0). I created a temporary fresh profile, but there are still no icons. It seems from some reddit posts that no icons is the norm now, the developer of the Lepton theme has posted how to bring them back (using userchrome). I am happy using the userchrome I posted earlier and I'll likely add the missing icons one at a time, but I was curious what you had used - I'm guessing you don't know what is different in your case.

**Comment by jabcreations** [comment]
Rob, I'm thinking it might be the combination of you (possibly using) Linux and some oddball setting with the internal browser's theme. I'm using Windows 10 x64 and Waterfox 6.5/128 ESR is using the "System" theme. I don't know if Lepton is a Linux or browser theme?

**Comment by robbely** [comment]
> Rob, I'm thinking it might be the combination of you (possibly using) Linux and some oddball setting with the internal browser's theme. I'm using Windows 10 x64 and Waterfox 6.5/128 ESR is using the "System" theme. I don't know if Lepton is a Linux or browser theme?

I am using the same as you Windows 10 x64 and Waterfox 6.5.1. The issue seems to be a Firefox one, not a Waterfox one (Waterfox just inherits it) and appears to be intentional. Here are a few articles I found:

https://www.reddit.com/r/firefox/comments/1fez9e3/how_is_it_that_firefox_desktop_still_doesnt_use/
https://www.reddit.com/r/waterfox/comments/wt484g/how_to_removedisablehide_menu_and_menu_bar_icons/
https://news.ycombinator.com/item?id=39428411

Lepton, I believe was a previous built-in default theme for icons/appearance etc for Firefox/Waterfox etc that now has to be installed manually.

tl;dr - if you are using an unmodified/default installation (with new profile) of the latest Firefox ESR or Waterfox, then you should have missing icons too.

**Comment by jabcreations** [comment]
Are you seriously complaining about an issue while using Firefox that doesn't occur in Waterfox on the Waterfox Git? 🤨

I just created a new Waterfox profile for Waterfox 6.5 and the icons appear.

**Comment by robbely** [comment]
> Are you seriously complaining about an issue while using Firefox that doesn't occur in Waterfox on the Waterfox Git? 🤨
> 
> I just created a new Waterfox profile for Waterfox 6.5 and the icons appear.

I wasn't complaining, as I said previously I'm happy with the fix someone else posted elsewhere, I was just curious if you had done something different. I'm not sure why you're getting icons when they are supposed to have disappeared, but I'm glad we've both got them working in our own ways.

**Comment by jabcreations** [comment]
Either I misread or you mistyped though it sounds like you've got them working which is great!

**Comment by spykos** [comment]
I just created a new profile in windows and the restart button is still not shown.

**Comment by robbely** [comment]
> I just created a new profile in windows and the restart button is still not shown.

It hasn't been fixed yet, but it has been added to the official list of known issues. 
jabcreations has a fix posted above you can use for now until it is officially fixed.

**Comment by SpoopyTim** [comment]
I'd just like to point out, there's an alternative way to restart without the menu button. It's what I'm using until there's an official fix.

Just go to `about:profiles` and hit `Restart normally`

<img width="1899" alt="Image" src="https://github.com/user-attachments/assets/7f1dcade-5660-4f21-94e3-02d31327753b" />

**Comment by aermak** [comment]
just pin the about:profiles page to have a poor man's restart shortcut in 2 clicks just like before!💡

**Comment by rn5577** [comment]
You can use a Firefox extension called 'Restart Button'. It works for me & can be used for those of a non-tech mind.

**Comment by ghost** [comment]
> You can use a Firefox extension called 'Restart Button'. It works for me & can be used for those of a non-tech mind.

Is that the one with the average score of 2.2 stars?

It doesn't actually restart the browser unless you buy the "Time Restart" extension.

No thanks.
